sample-app/src/utils/geolocationutils.tsx

```diff
@@ -0,0 +1,56 @@
+import { AnswersActions } from "@yext/answers-headless-react";
+import { SearchIntent } from "@yext/answers-headless";
+
+const defaultGeolocationOptions: PositionOptions = {
+  enableHighAccuracy: false,
+  timeout: 6000,
+  maximumAge: 300000,
+};
+
+/**
+ * Executes universal or vertical search with user location retrieved from nagivator.geolocation API.
+ * If useLocationOnNearMeOnly is true, will attempt to get user location only if the query's searchIntents
+ * (retrieved through autocomplete response) contains 'NEAR_ME'. If there's a failure in getting
+ * user location, a normal search is performed.
+ */
+export async function executeSearchWithUserLocation(
+  answersActions: AnswersActions,
+  isVertical: boolean,
+  geolocationOptions?: PositionOptions,
+  useLocationOnNearMeOnly?: boolean
+) {
+  const executeSearch = () => {
+    isVertical
+      ? answersActions.executeVerticalQuery()
+      : answersActions.executeUniversalQuery();
+    return;
+  }
+
+  if (useLocationOnNearMeOnly) {
+    const results = isVertical
+      ? await answersActions.executeVerticalAutocomplete()
+      : await answersActions.executeUniversalAutocomplete();
+    if (!results?.inputIntents?.includes(SearchIntent.NearMe)) {
```

Comment:
as somebody is typing queries into the searchbar, each key stroke will already fire off an autocomplete request. Would it be possible to use the inputIntents from there instead?

Reply:
that's not stored in headless state, but rather inside searchbar component state. it would be tricky to pass that around to different components. Maybe the searchIntent of autocomplete should be store somewhere, or have the ability to update the searchIntent in QueryState? what do you think

Reply:
since the search bar executes autocomplete requests, would it be possible to pass along the input intents? Maybe by awaiting the autocomplete request? I had a potentially jank idea where we store the most recent autocomplete request's promise as a ref, and always await it before making searches.

Reply:
It would get us out of making duplicate autocomplete requests before every search, though. I wouldn't be surprised if there's also a nicer way to implement it.

Reply:
We could pass in the inputIntents as a param to this method, so that cases like searching on page load will still fire the autocomplete to check for "near me"

Reply:
oh for some reason, I was thinking vertical/universal search page wouldn't be able to get intents from search bar for the default search, but I forgot since it's an initial search, we would execute an autocomplete request anyway for the intents. As for searchbar, I think that idea would work. I will something out

Reply:
updated!